### DISCUSSION TITLE
Failed to send hostname reply: Invalid argument

### DISCUSSION THREAD

**Discussion by sudhanshubhasin** [ltsp/ltsp#848] [General]
LTSP server and clients keep loosing interest and found the logs via journalctl for repeat of the below instances every 20-30 mins

Jul 31 13:50:13 users09 dbus-daemon[768]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Jul 31 13:50:13 users09 systemd[1]: Started Network Manager Script Dispatcher Service.
Jul 31 13:50:14 users09 NetworkManager[770]: <info>  [1690791614.2839] manager: NetworkManager state is now CONNECTED_GLOBAL
Jul 31 13:50:14 users09 whoopsie[1683]: [13:50:14] The default IPv4 route is: /org/freedesktop/NetworkManager/ActiveConnection/1
Jul 31 13:50:14 users09 whoopsie[1683]: [13:50:14] Not a paid data plan: /org/freedesktop/NetworkManager/ActiveConnection/1
Jul 31 13:50:14 users09 whoopsie[1683]: [13:50:14] Found usable connection: /org/freedesktop/NetworkManager/ActiveConnection/1
Jul 31 13:50:14 users09 whoopsie[1683]: [13:50:14] online
Jul 31 13:50:24 users09 systemd[1]: NetworkManager-dispatcher.service: Succeeded.
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0666] dhcp4 (eno1): option dhcp_lease_time      => '7200'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0667] dhcp4 (eno1): option domain_name_servers  => '192.168.0.1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0667] dhcp4 (eno1): option expiry               => '1690799078'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0667] dhcp4 (eno1): option host_name            => 'users09'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0668] dhcp4 (eno1): option ip_address           => '192.168.0.6'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0668] dhcp4 (eno1): option next_server          => '192.168.0.1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0668] dhcp4 (eno1): option requested_broadcast_address => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0668] dhcp4 (eno1): option requested_domain_name => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0668] dhcp4 (eno1): option requested_domain_name_servers => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0669] dhcp4 (eno1): option requested_domain_search => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0669] dhcp4 (eno1): option requested_host_name  => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0669] dhcp4 (eno1): option requested_interface_mtu => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0669] dhcp4 (eno1): option requested_ms_classless_static_routes => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0669] dhcp4 (eno1): option requested_nis_domain => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0670] dhcp4 (eno1): option requested_nis_servers => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0670] dhcp4 (eno1): option requested_ntp_servers => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0670] dhcp4 (eno1): option requested_rfc3442_classless_static_routes => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0670] dhcp4 (eno1): option requested_root_path  => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0671] dhcp4 (eno1): option requested_routers    => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0671] dhcp4 (eno1): option requested_static_routes => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0671] dhcp4 (eno1): option requested_subnet_mask => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0671] dhcp4 (eno1): option requested_time_offset => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0671] dhcp4 (eno1): option requested_wpad       => '1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0672] dhcp4 (eno1): option routers              => '192.168.0.1'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0672] dhcp4 (eno1): option subnet_mask          => '255.255.255.0'
Jul 31 13:54:38 users09 NetworkManager[770]: <info>  [1690791878.0672] dhcp4 (eno1): state changed extended -> extended
Jul 31 13:54:38 users09 dbus-daemon[768]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service' requested by ':1.12' (uid=0 pid=770 comm="/usr/sbin/NetworkManager --no-daemon " label="unconfined")
Jul 31 13:54:38 users09 systemd[1]: Starting Network Manager Script Dispatcher Service...
Jul 31 13:54:38 users09 dbus-daemon[768]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Jul 31 13:54:38 users09 systemd[1]: Started Network Manager Script Dispatcher Service.
Jul 31 13:54:48 users09 systemd[1]: NetworkManager-dispatcher.service: Succeeded.
Jul 31 13:56:16 users09 gnome-shell[3174]: [3167:3196:0731/135616.755136:ERROR:cert_issuer_source_aia.cc(34)] Error parsing cert retrieved from AIA (as DER):
Jul 31 13:56:16 users09 gnome-shell[3174]: ERROR: Couldn't read tbsCertificate as SEQUENCE
Jul 31 13:56:16 users09 gnome-shell[3174]: ERROR: Failed parsing Certificate
Jul 31 13:56:16 users09 gnome-shell[3174]: [3167:3196:0731/135616.758436:ERROR:cert_verify_proc_builtin.cc(702)] CertVerifyProcBuiltin for 192.168.0.179 failed:
Jul 31 13:56:16 users09 gnome-shell[3174]: ----- Certificate i=0 (CN=cloudecomr.synology.me) -----
Jul 31 13:56:16 users09 gnome-shell[3174]: ERROR: Time is after notAfter
Jul 31 13:56:16 users09 gnome-shell[3174]: [3221:3239:0731/135616.758757:ERROR:ssl_client_socket_impl.cc(978)] handshake failed; returned -1, SSL error code 1, net_error -200
Jul 31 13:56:16 users09 gnome-shell[3174]: [3221:3239:0731/135616.760167:ERROR:ssl_client_socket_impl.cc(978)] handshake failed; returned -1, SSL error code 1, net_error -200
Jul 31 14:05:13 users09 systemd-resolved[116410]: Failed to send hostname reply: Invalid argument
Jul 31 14:05:13 users09 NetworkManager[770]: <info>  [1690792513.8425] manager: NetworkManager state is now CONNECTED_SITE
Jul 31 14:05:13 users09 whoopsie[1683]: [14:05:13] offline
Jul 31 14:05:13 users09 dbus-daemon[768]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service' requested by ':1.12' (uid=0 pid=770 comm="/usr/sbin/NetworkManager --no-daemon " label="unconfined")
Jul 31 14:05:13 users09 systemd[1]: Starting Network Manager Script Dispatcher Service...
Jul 31 14:05:13 users09 dbus-daemon[768]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Jul 31 14:05:13 users09 systemd[1]: Started Network Manager Script Dispatcher Service.
Jul 31 14:05:19 users09 systemd-resolved[116410]: Failed to send hostname reply: Invalid argument
Jul 31 14:05:19 users09 systemd-resolved[116410]: Failed to send hostname reply: Invalid argument
Jul 31 14:05:19 users09 whoopsie[1683]: [14:05:19] Cannot reach: https://daisy.ubuntu.com
Jul 31 14:05:19 users09 systemd-resolved[116410]: Failed to send hostname reply: Invalid argument
Jul 31 14:05:23 users09 systemd[1]: NetworkManager-dispatcher.service: Succeeded.
Jul 31 14:05:24 users09 systemd-resolved[116410]: Failed to send hostname reply: Invalid argument
Jul 31 14:05:29 users09 systemd-resolved[116410]: Failed to send hostname reply: Invalid argument
Jul 31 14:05:45 users09 systemd-resolved[116410]: Failed to send hostname reply: Invalid argument
Jul 31 14:06:18 users09 NetworkManager[770]: <info>  [1690792578.2413] manager: NetworkManager state is now CONNECTED_GLOBAL
Jul 31 14:06:18 users09 dbus-daemon[768]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service' requested by ':1.12' (uid=0 pid=770 comm="/usr/sbin/NetworkManager --no-daemon " label="unconfined")
Jul 31 14:06:18 users09 whoopsie[1683]: [14:06:18] The default IPv4 route is: /org/freedesktop/NetworkManager/ActiveConnection/1
Jul 31 14:06:18 users09 systemd[1]: Starting Network Manager Script Dispatcher Service...
Jul 31 14:06:18 users09 whoopsie[1683]: [14:06:18] Not a paid data plan: /org/freedesktop/NetworkManager/ActiveConnection/1
Jul 31 14:06:18 users09 whoopsie[1683]: [14:06:18] Found usable connection: /org/freedesktop/NetworkManager/ActiveConnection/1
Jul 31 14:06:18 users09 dbus-daemon[768]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Jul 31 14:06:18 users09 systemd[1]: Started Network Manager Script Dispatcher Service.
Jul 31 14:06:19 users09 whoopsie[1683]: [14:06:19] online
Jul 31 14:06:28 users09 systemd[1]: NetworkManager-dispatcher.service: Succeeded.

**Comment by alkisg** [comment]
Hi, issues are for bugs in the LTSP code, this is probably just a miconfiguration or a networking issue somewhere.
Note that the LTSP client network connection is NOT handled by NetworkManager.
Converting into a discussion.

**Reply by sudhanshubhasin** [reply]
No networking issue present and other servers facing public are absolutely working flawless

**Reply by sudhanshubhasin** [reply]
Also to note - both LTSP Server and LTSP  Clients loose internet access and generate these logs almost every 20-30 mins

**Comment by alkisg** [comment]
The LTSP code doesn't handle the server Internet connections. So whatever problem it has and gets disconnected, LTSP is not to blame...
Try non-LTSP related things to pinpoint the issue. E.g. try to work on your server while no LTSP client is booted. Or try to boot your LTSP server with a live CD and see if the problem happens there as well.
After you stabilize your LTSP server connection, *then* start checking on your LTSP clients.

**Reply by sudhanshubhasin** [reply]
Non LTSP system has no problem - it is only the LTSP Server and the clients have the issues
I will try not booting LTSP clients to find if LTSP server has no problem
